feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.adapter.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from '@angular/core';
+import { CartModification } from '@spartacus/core';
+import { CommonConfigurator } from '@spartacus/product-configurator/common';
+import { Observable, of, throwError } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { RulebasedConfiguratorAdapter } from '../core/connectors/rulebased-configurator.adapter';
+import { Configurator } from '../core/model/configurator.model';
+import { CpqConfiguratorOccService } from '../occ/cpq/cpq-configurator-occ.service';
+import { CpqConfiguratorRestService } from './cpq-configurator-rest.service';
+
+@Injectable()
+export class CpqConfiguratorRestAdapter
+  implements RulebasedConfiguratorAdapter {
+  constructor(
+    protected cpqRestService: CpqConfiguratorRestService,
+    protected cpqOccService: CpqConfiguratorOccService
+  ) {}
+
+  getConfiguratorType(): string {
+    return 'CLOUDCPQCONFIGURATOR';
+  }
+
+  createConfiguration(
+    owner: CommonConfigurator.Owner
+  ): Observable<Configurator.Configuration> {
+    if (!owner.id) {
```

Comment:
@ChristophHi didn't we discuss that id (or key?) should be a mandatory field of the `Owner` interface? Then the check here could be removed from the adapter.

Reply:
No, id will still be optional. 'key' is this essential one